starlette/middleware/errors.py

```diff
@@ -158,20 +158,23 @@ async def _send(message: Message) -> None:
         try:
             await self.app(scope, receive, _send)
         except Exception as exc:
+            request = Request(scope)
+
+            if self.handler:
+                # Use an installed 500 error handler.
+                if asyncio.iscoroutinefunction(self.handler):
+                    response = await self.handler(request, exc)
+                else:
+                    response = await run_in_threadpool(self.handler, request, exc)
+
+            # This flow overrides the former response, if there was any.
             if not response_started:
```

Comment:
I don't really understand the change footprint here.

From the discussion I would have thought that the change would be to move everything outside the `if not response started` block *except* `await response(scope, receive, send)`.

(Ie. keep the behaviour the same, except for the fact that we *always* run the handler)

Reply:
You mean like this:

```python
            request = Request(scope)
            if self.debug:
                # In debug mode, return traceback responses.
                response = self.debug_response(request, exc)
            elif self.handler is None:
                # Use our default 500 error handler.
                response = self.error_response(request, exc)
            else:
                # Use an installed 500 error handler.
                if asyncio.iscoroutinefunction(self.handler):
                    response = await self.handler(request, exc)
                else:
                    response = await run_in_threadpool(self.handler, request, exc)

            if not response_started:
                await response(scope, receive, send)
```
?

Reply:
I can do that :+1: 

It doesn't make a practical difference, as the first two conditionals don't perform any logic besides sending back a response object, but I agree it's cleaner.

Reply:
Great, yup. I think it's easier to understand that way round. (And it's a more obvious review too.)